README.md

```diff
@@ -1,58 +1,2501 @@
-# TODO
+# IBM Cloud Kubernetes Service open source Ingress controller
 
-### Building
+TODO short description here - do we want to mentin anything about how this is now officially unsupported so it was instead open sourced?
 
-To build on a Mac or Linux based machine, ensure Docker is running and run:
-`make container`
+## Prerequisites and considerations
+Before you get started with the IBM Cloud Kubernetes Service open source Ingress controller, review the following prerequisites and considerations.
 
+TODO: do these general IKS prereqs still apply? i would think they would for any Ingress controller in an iks cluster
 
-### Pushing to IBM Cloud Registry
-* Find IBM Cloud Registry documentation here: https://cloud.ibm.com/docs/Registry?topic=Registry-registry_overview
+- Setting up Ingress requires the following [IBM Cloud IAM roles](cloud.ibm.com/docs/containers?topic=containers-users#platform):
+    - **Administrator** platform access role for the cluster
+    - **Manager** service access role in all namespaces
+- If you restrict network traffic to edge worker nodes, ensure that at least two [edge worker nodes](cloud.ibm.com/docs/containers?topic=containers-edge) are enabled in each zone so that Ingress controllers deploy uniformly.
+- To be included in Ingress load balancing, the names of the `ClusterIP` services that expose your apps must be unique across all namespaces in your cluster.
+- VPC clusters: [Allow traffic requests that are routed by Ingress to node ports on your worker nodes](cloud.ibm.com/docs/containers?topic=containers-vpc-network-policy#security_groups).
+- To build on a Mac or Linux based machine, ensure Docker is running and run: `make container`
 
-* Login to the IBM Container Registry
-`ibmcloud login`
-`ibmcloud cr login`
+TODO: need a warning re: this isnt support by iks, you must manage, etc
```

Comment:
Are you asking that I put a warning message that gets echoed from the Makefile when the user builds?